client/src/hooks/datafiles/mutations/useTrash.ts

```diff
@@ -0,0 +1,127 @@
+import { useSelector, useDispatch, shallowEqual } from 'react-redux';
+import { useMutation } from '@tanstack/react-query';
+import { useSelectedFiles } from 'hooks/datafiles';
+import Cookies from 'js-cookie';
+
+export async function trashUtil({
+  api,
+  scheme,
+  system,
+  path,
+  homeDir,
+}: {
+  api: string;
+  scheme: string;
+  system: string;
+  path: string;
+  homeDir: string;
+}) {
+  const url = `/api/datafiles/${api}/trash/${scheme}/${system}/${path}/`;
+  // const request = await apiClient.put(url, body, {
+  //   headers: {
+  //     'X-CSRFToken': Cookies.get('csrftoken' || ''),
+  //   },
+  //   withCredentials: true,
+  // });
+  const request = await fetch(url, {
+    method: 'PUT',
+    headers: { 'X-CSRFToken': Cookies.get('csrftoken') || ''},
+    credentials: 'same-origin',
+    body: JSON.stringify({
+      homeDir: homeDir,
+    }),
+  });
+  // const request = await apiClient.put(url, body);
+  console.log(request);
+  
+  return request;
+}
+
+function useTrash() {
+  const dispatch = useDispatch();
+  const { selectedFiles: selected } = useSelectedFiles();
+  const status = useSelector(
+    (state: any) => state.files.operationStatus.trash,
+    shallowEqual
+  );
+
+  const { api, scheme } = useSelector(
+    (state: any) => state.files.params.FilesListing
+  );
+
+  const setStatus = (newStatus: any) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: newStatus, operation: 'trash' },
+    });
+  };
+
+  const { mutateAsync } = useMutation({ mutationFn: trashUtil });
+
+  const trash = ({
+    destSystem,
+    homeDir,
+    callback,
+  }: {
+    destSystem: string;
+    homeDir: any;
+    callback: any;
+  }) => {
+    const filteredSelected = selected.filter(
+      (f: any) => status[f.id] !== 'SUCCESS'
+    );
+    const trashCalls: Promise<any>[] = filteredSelected.forEach((file: any) => {
+      dispatch({
+        type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
+        payload: {
+          status: 'RUNNING',
+          key: (index: string) => index,
+          operation: 'trash',
+        },
+      });
+      console.log("File", file);
+      return mutateAsync(
+        {
+          api: api,
+          scheme: scheme,
+          system: destSystem,
+          path: file.path,
+          homeDir: homeDir,
+        },
+        {
```

Comment:
We're going to ensure that all deleted files each have a promise associated with them. Instead of having a 1 promise for the whole map. See [this](https://github.com/TACC/Core-Portal/blob/51e2230f3fca51a1e39104b3a7216d0d154d8f38/client/src/hooks/datafiles/mutations/useUpload.ts#L101) for how it uses Promise all then sends the dispatches out when all promises are met.

Reply:
I just pushed up the changes.